### PR TITLE
DOC: Group relevant df.mask/df.where examples together

### LINKED ISSUE #25187: DOC: DataFrame.mask() uses df.where in the examples section
In the stables doc page of `pandas.DataFrame.mask()` in the bottom section of 'Examples', is showing executions of using `pandas.DataFrame.where()` method instead of mask. I refer to this [page](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.mask.html)

### PR DESCRIPTION
**Pull request by hongshaoyang**
df.mask() and df.where() are complementary and so they share docs, see [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.mask.html). I grouped related examples together.

- [x] Close #25187
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -9104,6 +9103,13 @@ def where(
         3    3
         4    4
         dtype: int64
+        >>> s.mask(s > 1, 10)
```
Comment (jreback) [inline pandas/core/generic.py:9106]: can you add a blank line, maybe worth adding comment that mask and where are inverse selections

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (hongshaoyang) [reply pandas/core/generic.py:9106]: I chose not to add a blank line because the example above is the corresponding `s.where(s > 1, 10)` example

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @hongshaoyang

**Comment by hongshaoyang** [comment]
Thanks for the merge! May I get this PR labelled as `hacktoberfest-accepted`?

See #36837 and [here](https://hacktoberfest.digitalocean.com/hacktoberfest-update).

**Pull request merged**